var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -475,33 +493,29 @@ def cmake_args(self):
         # Same applies for raja
         if self.spec.satisfies("+omp~kokkos~raja"):
             args.append("-DCMAKE_C_COMPILER=" + spack_cc)
-            if "cuda_arch" in self.spec.variants:
-                # the architecture value is only number so append cc_ to the name
-                cuda_arch = "cc" + self.spec.variants["cuda_arch"].value[0]
+            if self.spec.satisfies("~omp_offload"):
+                args.append("-DOFFLOAD=" + "OFF")
+                # Check if the omp_flags variant is not set to "none"
+                args.append("-DCMAKE_CXX_FLAGS=" 
+                + self.pkg.compiler.openmp_flag 
+                + " " + (self.spec.variants["omp_flags"].value if self.spec.variants["omp_flags"].value != "none" else ""))
+            else:
+                offload_args=""
                 args.append("-DOFFLOAD=ON")
-                args.append("-DOFFLOAD_FLAGS=" + " -mp=gpu;" + "-gpu=" + cuda_arch)
-            elif ("amdgpu_target" in self.spec.variants) and (
+                if "cuda_arch" in self.spec.variants:
+                    if self.spec.satisfies("%nvhpc"):
+                        cuda_arch = "cc" + self.spec.variants["cuda_arch"].value[0]
+                        offload_args=" -mp=gpu;" + "-gpu=" + cuda_arch + " "
+                    if self.spec.satisfies("%clang"):
+                        cuda_arch = "sm_" + self.spec.variants["cuda_arch"].value[0]
+                        offload_args="-march=znver3;-fopenmp;--offload-arch="+cuda_arch
```

Comment:
From CMake this was supplied `-march=native` but seems like the package didn't like this, how could we force `march` to be the CPU arch we have in the system ?  @tgamblin @becker33